### PR TITLE
Add ActionClient.Watch to watch progress of actions

### PR DESCRIPTION
**Pull request by thetechnick**

### CODE REVIEW

File: hcloud/action.go [hunk 1]
```diff
@@ -125,3 +125,62 @@ func (c *ActionClient) All(ctx context.Context) ([]*Action, error) {
 
 	return allActions, nil
 }
+
+// Wait watches the action state until it completes with success or error.
+// Also provides progress updates on the first channel.
+func (c *ActionClient) Wait(ctx context.Context, action *Action) (<-chan int, <-chan error) {
```
Comment (thcyron) [inline hcloud/action.go]: I don’t think Wait is a good name because it does not wait. Maybe Watch or WatchProgress?

File: hcloud/action.go [hunk 2]
```diff
@@ -125,3 +125,62 @@ func (c *ActionClient) All(ctx context.Context) ([]*Action, error) {
...
+		sendProgress := func(p int) {
+			select {
+			case progressCh <- p:
+				break
+			default:
+				break
+			}
+		}
+
+		var retries = 0
```
Comment (thcyron) [inline hcloud/action.go]: What about resetting it to 0 after a successful response?

File: hcloud/action_test.go [hunk 3]
```diff
@@ -134,3 +134,76 @@ func TestActionClientAll(t *testing.T) {
...
+		select {
+		case progress := <-progressCh:
+			progressUpdates = append(progressUpdates, progress)
+		case err = <-errCh:
+			break loop
+		}
+	}
+
+	if err == nil {
+		t.Fatalf("error expected %v", err)
```
Comment (thcyron) [inline hcloud/action_test.go]: Wrong message?

File: hcloud/action_test.go [hunk 4]
```diff
@@ -134,3 +134,76 @@ func TestActionClientAll(t *testing.T) {
...
+	}
+	if len(progressUpdates) != 1 && progressUpdates[0] != 50 {
```
Comment (thcyron) [inline hcloud/action_test.go]: Should be `||`?

File: hcloud/action.go (hunk 2 above)
Comment (thetechnick) [reply hcloud/action.go]: done

File: hcloud/action_test.go (hunk 3 above)
Comment (thetechnick) [reply hcloud/action_test.go]: Is correct because i expect an non nil `err` because the state of the action is error

Comment (thcyron) [reply hcloud/action_test.go]: Your condition might be correct, but why would you print `error expected <nil>`?

**Pull request merged**